scripts/push_sdk.sh

```diff
@@ -1,94 +1,144 @@
+  exit 1
 fi
 
+setup_ssh() {
+  eval "$(ssh-agent -s)"
+  ssh-add - <<< "${SDK_REPO_PRIVATE_KEY}"
+}
+
 git_clone() {
-  git clone --depth 1 https://${GH_TOKEN}@github.com/$1.git
+  echo "[INFO] Cloning $ORGANIZATION_NAME/$REPOSITORY_NAME repository..."
+
+  cd $SDK_REPO_DIR
+  git clone --depth 1 git@github.com:$ORGANIZATION_NAME/$REPOSITORY_NAME.git
+  SDK_REPO_DIR="$SDK_REPO_DIR/$REPOSITORY_NAME"
+
+  echo "[INFO] Success. Repository cloned at $SDK_REPO_DIR"
+}
+
+remove_previous_sdks() {
+  echo "[INFO] Removing previous SDKs..."
+
+  sdks_directory="$SDK_REPO_DIR/sdks/$LANGUAGE"
+
+  if [[ -d $sdks_directory ]]; then
+    cd $sdks_directory
+    rm -rf *
+    echo "[INFO] Success."
+  else
+    echo "[WARN] Directory $REPO_NAME/sdks doesn't exists, skipping."
+  fi
+}
+
+copy_new_sdks() {
+  echo "[INFO] Copying new SDKs..."
+
+  sdks_directory="$SDK_REPO_DIR/sdks"
+
+  if [[ ! -d $sdks_directory ]]; then
+    echo "[WARN] Directory $sdks_directory doesn't exists, creating it..."
+    mkdir $sdks_directory
+    echo "[INFO] Directory $sdks_directory created."
+  fi
+
+  cp -r "$GENERATOR_REPO_DIR/generated-sources/$LANGUAGE/." "$SDK_REPO_DIR/sdks"
+
+  echo "[INFO] Copy successful."
 }
 
 setup_git() {
-  git config user.email $github.actor
-  git config user.name "GITHUB CI"
+  echo "[INFO] Setting up GH credentials..."
+
+  git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+  git config user.name "$GITHUB_ACTOR"
+
+  echo "[INFO] Success. Email: $GITHUB_ACTOR@users.noreply.github.com, Name: $GITHUB_ACTOR"
 }
 
 git_add_files() {
-  GIT_TRACK_FILE="${SCRIPT_ROOT}/git-track/${1}.txt"
-  if [[ -f ${GIT_TRACK_FILE} ]]; then
-      files_to_add=$(cat ${GIT_TRACK_FILE} | grep -Ev "^\s*#|^\s*$" | tr '\n' ' ')
-      git add ${files_to_add}
-  else
-    echo "'${GIT_TRACK_FILE}' does not exists, no files to add."
-    exit 1
-  fi
+  cd $SDK_REPO_DIR
+  git add .
 }
 
 git_commit_and_tag() {
-  version=$1
-  if [[ ${version} == "" ]]; then
-    echo "version is not defined"
-    exit 1
-  fi
-  git commit -m "Automatic update of SDK - ${version}" && git tag ${version}
+  git commit -m "Automatic update of SDK - $TAG_VERSION"
+  git tag $TAG_VERSION
 }
 
 git_push() {
-  if [[ ${github.actor} == "github" ]]; then
-    git push origin --tags --quiet && git push origin --quiet
-  else
-    echo "Only user travis should be able to push."
-  fi
+  git push origin --quiet
+  # Push tag
+  git push origin --tags --quiet
 }
 
 process() {
-  language=$1
-  mkdir ${BUILD_DIR}/criteo
-  cd ${BUILD_DIR}/criteo
-  REPO="criteo/criteo-api-${language}-sdk"
-  git_clone ${REPO}
+  setup_ssh
 
-  NEXT_CLIENT="NEXT_criteo-api-${language}-sdk"
-  cp -r "${SCRIPT_ROOT}/../generated-sources/${language}" ${NEXT_CLIENT}
-  cp -r "${BUILD_DIR}/${REPO}/.git" ${NEXT_CLIENT}
+  git_clone
 
-  cd ${NEXT_CLIENT}
+  remove_previous_sdks
 
-  # add files before doing the diff
-  git_add_files ${language}
+  copy_new_sdks
+
+  git_add_files
```

Comment:
should we continue if copy_new_sdks failed (or the script will stop by itself?)

Reply:
the first command line of the script, `set -ex`, should prevent this from happening, as it should stop the script if any simple command fails